### PR TITLE
Load/save cleanup

### PR DESCRIPTION
**Pull request by vlolteanu**

### CODE REVIEW

**Review by glebm — APPROVED** [review]

File: Source/loadsave.cpp [hunk 1]
```diff
@@ -128,25 +128,20 @@ class LoadHelper {
...
 
 public:
 	SaveHelper(const char *szFileName, size_t bufferLen)
+		: m_szFileName(szFileName)
+		, m_buffer(std::make_unique<uint8_t[]>(codec_get_encoded_len(bufferLen)))
+		, m_capacity(bufferLen)
 	{
-		m_szFileName = szFileName;
-		m_capacity = bufferLen;
-		m_buffer = DiabloAllocPtr(codec_get_encoded_len(m_capacity));
```
Comment (AJenbo) [inline Source/loadsave.cpp:149]: What is the benifit of doing the assignment in the signature instead of the body?

Comment (vlolteanu) [reply Source/loadsave.cpp:149]: In terms of practicality, none.

Comment (AJenbo) [reply Source/loadsave.cpp:149]: typical C++ to have an alternative syntax for things

**Comment by AJenbo** [comment]
looks like this is causing master to fail on switch, is that something you could look into?

**Comment by vlolteanu** [comment]
What do you mean by master failing on switch?

**Comment by AJenbo** [comment]
the CI failed to build the Nintendo switch version https://app.circleci.com/pipelines/github/diasurgical/devilutionX/5283/workflows/54f8113e-8c71-4fe8-8940-c5fb807a7708/jobs/38100

but turns out it was an issue with the server, everything is fine

**Pull request merged**